### PR TITLE
feat: Add new SIG Config implementations for Azure Linux CgroupV2 images

### PR DESCRIPTION
**Pull request by henryli001**
**What type of PR is this?**

**What this PR does / why we need it**:
This PR adds new resource groups and galleries to host the new Azure Linux CgroupV2 images, separating from the existing ACG that contains all the CgroupV1 Azure Linux images.

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Comment by cameronmeissner** [comment]
lgtm I think, lets wait on the RP-side changes to merge in first though

File: pkg/agent/datamodel/sig_config.go [hunk 1]
```diff
@@ -495,55 +526,111 @@ var (
 		Version:       LinuxSIGImageVersion,
 	}
 
-	SIGCBLMarinerV2ImageConfigTemplate = SigImageConfigTemplate{
+	SIGAzureLinuxV2Gen1ImageConfigTemplate = SigImageConfigTemplate{
+		ResourceGroup: AKSAzureLinuxResourceGroup,
+		Gallery:       AKSAzureLinuxGalleryName,
+		Definition:    "V2",
+		Version:       LinuxSIGImageVersion,
```
Comment (anujmaheshwari1) [inline pkg/agent/datamodel/sig_config.go:533]: this needs to change to 202309.11.0, once this image is tested e2e through RP, then we can sync with regular sig images.

File: pkg/agent/baker.go [hunk 2]
```diff
@@ -474,7 +474,7 @@ func getContainerServiceFuncMap(config *datamodel.NodeBootstrappingConfiguration
 		},
 		"IsMariner": func() bool {
 			// TODO(ace): do we care about both? 2nd one should be more general and catch custom VHD for mariner
-			return profile.Distro.IsCBLMarinerDistro() || isMariner(config.OSSKU)
+			return profile.Distro.IsAzureLinuxDistro() || isMariner(config.OSSKU)
```
Comment (anujmaheshwari1) [inline pkg/agent/baker.go:477]: apologies, is this func already defined? I cant seem to find it in types.go

File: pkg/agent/datamodel/sig_config.go (hunk 1 above)
Comment (henryli001) [reply pkg/agent/datamodel/sig_config.go]: Fixed

File: pkg/agent/baker.go (hunk 2 above)
Comment (anujmaheshwari1) [reply pkg/agent/baker.go:477]: my bad, its there

File: pkg/agent/datamodel/sig_config.go [hunk 3]
```diff
@@ -495,55 +526,111 @@ var (
...
 		Gallery:       AKSCBLMarinerGalleryName,
 		Definition:    "V2katagen2TL",
 		Version:       CBLMarinerV2KataGen2TLSIGImageVersion,
 	}
 
+	SIGAzureLinuxV2KataGen2TLImageConfigTemplate = SigImageConfigTemplate{
+		ResourceGroup: AKSAzureLinuxResourceGroup,
+		Gallery:       AKSAzureLinuxGalleryName,
+		Definition:    "V2katagen2TL",
+		Version:       CBLMarinerV2KataGen2TLSIGImageVersion,
```
Comment (anujmaheshwari1) [inline pkg/agent/datamodel/sig_config.go:631]: we didnt actually release katagen2TL image for cgroupsv2, maybe we can remove this?

**Review by anujmaheshwari1 — APPROVED** [review]
lgtm!

File: pkg/agent/datamodel/sig_config.go (hunk 3 above)
Comment (henryli001) [reply pkg/agent/datamodel/sig_config.go]: Fixed

**Pull request merged**